### PR TITLE
Add tensorboardX recipe

### PR DESCRIPTION
**Pull request by mdraw**
Packages https://github.com/lanpa/tensorboard-pytorch, a library for direct tensorboard logging of NumPy arrays, PyTorch tensors etc. without TensorFlow or TensorBoard requirements.

Notes:
- The repository is still called tensorboard-pytorch, but the project name has been changed
- As far as I see, the log format of tensorboardX is not compatible with recent versions of the official TensorBoard server https://github.com/tensorflow/tensorboard, but only with https://github.com/dmlc/tensorboard, which has a name conflict with Google's Tensorflow (see https://github.com/dmlc/tensorboard/issues/50) and is not packaged on conda-forge. I am not sure how we should deal with this issue. If we decided to also package dmlc/tensorboard, how should we name it?

### CODE REVIEW

File: recipes/tensorboardx/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    - python
+    - setuptools
+  run:
+    - python
+    - numpy
+    - protobuf
+
+test:
+  imports:
+    - tensorboardX
```
Comment (CJ-Wright) [inline recipes/tensorboardx/meta.yaml]: Please add more tests. (more imports are fine)

**Comment by mdraw** [comment]
All submodules are now imported at test time. Is it okay now?

**Comment by CJ-Wright** [comment]
This is noarch, and thus only needs to pass on circle.

**Comment by lanpa** [comment]
@mdraw I always test with tensorboardX with tensorflow-tensorboard. Can you provide more information about the compatibility? thanks

**Comment by mdraw** [comment]
I tried it again on a new Anaconda installation with tensorflow-tensorboard and it works now :+1:. There was probably an issue with my previous install that I can't reproduce.

**Pull request merged**